### PR TITLE
feat: use polkadot/polkadot-parachain patched

### PR DESCRIPTION
**Pull request by kratico**
Companion PR https://github.com/paritytech/capi-binary-builds/pull/36

Use the `polkadot` and `polkadot-parachain` patched version with 2s and 4s block times.

image

FYI, I did see an improvement in individual example runs but I did not see an improvement in the `deno task test:eg` timings.

### CODE REVIEW

File: capi.config.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { binary, Config } from "./mod.ts"
 
-const polkadot = binary("polkadot", "v0.9.38")
-const polkadotParachain = binary("polkadot-parachain", "v0.9.380")
+const polkadot = binary("polkadot", "v0.9.38-patched")
+const polkadotParachain = binary("polkadot-parachain", "v0.9.380-patched")
```
Comment (kratico) [inline capi.config.ts:4]: `v0.9.38-dev` might be better

WDYT?

**Comment by kratico** [comment]
> Any reason not to patch with 1s and 1s instead of 2s and 4s?

I did a couple of tests and there is no improvement with smaller values.
I guess, it's because there are more forks with lower values and it takes longer to finalize blocks.
Nevertheless, I think that babe/aura/grandpa pallets could be tweaked to get shorter block times.

**Comment by harrysolovay** [comment]
13s is still a win! Not quite instantaneous, but hopefully there's a path to get there.

File: server/devnetsHandler.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import * as f from "./factories.ts"
 const rDevnetsApi = /^\/devnets\/([\w-]+)$/
 
 export function createDevnetsHandler(
-  devnetTempDir: string,
+  devnetsTempDir: string,
```
Comment (harrysolovay) [inline server/devnetsHandler.ts:10]:
```suggestion
  devnetTempDir: string,
```

File: server/devnetsHandler.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export function createDevnetsHandler(
     if (net instanceof DevNetSpec) {
       const network = await networkMemo.run(
         net.relay,
-        () => net.relay.spawnNet(signal, devnetTempDir),
+        () => net.relay.spawnNet(signal, devnetsTempDir),
```
Comment (harrysolovay) [inline server/devnetsHandler.ts:34]:
```suggestion
        () => net.relay.spawnNet(signal, devnetTempDir),
```

**Review by harrysolovay — APPROVED** [review]

**Review by tjjfvi — APPROVED** [review]

**Pull request merged**